File: backend/internal/models/Transportation.go
package models

// Transportation représente les informations sur le transport pour un événement
type Transportation struct {
	ID         uint   `gorm:"primaryKey" json:"id"`
	EventID    uint   `json:"event_id" gorm:"not null"`
	UserID     uint   `json:"user_id"`
	Vehicle    string `json:"vehicle"`
	SeatNumber int    `json:"seat_number"`
}

type TransportationAdd struct {
	Email      string `json:"email"`
	Vehicle    string `json:"vehicle"`
	SeatNumber int    `json:"seat_number"`
}

File: client/lib/core/models/transportation.dart
class Transportation {
  final int id;
  final int seatNumber;
  final int eventId;
  final String vehicle;
  final int userId;
  final String email;

  Transportation({
    required this.id,
    required this.seatNumber,
    required this.eventId,
    required this.vehicle,
    required this.userId,
    required this.email,
  });

  factory Transportation.fromJson(Map<String, dynamic> json) {
    return Transportation(
      id: json['id'],
      seatNumber: json['seat_number'],
      eventId: json['event_id'],
      vehicle: json['vehicle'],
      userId: json['user_id'],
      email: json['email']  as String? ?? '',
    );
  }
}

class UserTransport {
  final int userId;
  final String firstname;
  final String lastname;
  final int participantId;
  final int eventId;
  final String email;
  final int transportationId;

  UserTransport({
    required this.userId,
    required this.firstname,
    required this.lastname,
    required this.participantId,
    required this.eventId,
    required this.email,
    required this.transportationId,
  });

  factory UserTransport.fromJson(Map<String, dynamic> json) {
    return UserTransport(
      userId: json['user_id'],
      firstname: json['firstname'],
      lastname: json['lastname'],
      participantId: json['participant_id'],
      eventId: json['event_id'],
      email: json['email'],
      transportationId: json['transportation_id'],
    );
  }
}